instrumentation/opentelemetry/github.com/gorilla/hypermux/mux.go

```diff
@@ -0,0 +1,38 @@
+package hypermux
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/hypertrace/goagent/instrumentation/opentelemetry"
+	sdkhttp "github.com/hypertrace/goagent/sdk/net/http"
+	otelhttp "go.opentelemetry.io/contrib/instrumentation/net/http"
+	"go.opentelemetry.io/otel/api/global"
+)
+
+func spanNameFormatter(operation string, r *http.Request) (spanName string) {
+	route := mux.CurrentRoute(r)
+	if route == nil {
+		return
+	}
+
+	var err error
+	spanName, err = route.GetPathTemplate()
```

Comment:
spanName is always nil when err is not? In that case it will be modified, no?

Reply:
if `err != nil`, spanName is `""`, still we need it as types must be respected.

Reply:
In that case to we want an empty spanName? Shouldn't we return the value that was passed in to this method?

Reply:
Good point. I will default to `request.Method`.

Reply:
ok.